demo/src/MemoryModels.tsx

```diff
@@ -1,55 +1,138 @@
-import React from "react";
-import { Button, Card, CardContent, TextField, Grid } from "@mui/material";
+import React, { useState } from "react";
+import {
+    Button,
+    Card,
+    CardContent,
+    Grid,
+    Input,
+    TextField,
+    Tooltip,
+    Typography,
+} from "@mui/material";
 
-type MemoryModelsUserInputPropTypes = {
-    onSubmit: (event: React.MouseEvent<HTMLFormElement>) => void;
-    setFormData: React.Dispatch<React.SetStateAction<string>>;
-    formData: string;
+type MemoryModelsInputPropTypes = {
+    setTextData: React.Dispatch<React.SetStateAction<string>>;
+    textData: string;
+};
+type MemoryModelsUserInputPropTypes = MemoryModelsInputPropTypes & {
+    onTextDataSubmit: (event: React.MouseEvent<HTMLFormElement>) => void;
 };
 
-export default function MemoryModelsUserInput(
-    props: MemoryModelsUserInputPropTypes
-) {
+function MemoryModelsFileInput(props: MemoryModelsInputPropTypes) {
+    const [uploadedFileString, setUploadedFileString] = useState("");
+
+    const onChange = (event) => {
+        try {
+            const uploadedFile = event.target.files[0];
+            const fileReader = new global.FileReader();
+            fileReader.readAsText(uploadedFile, "UTF-8");
+            fileReader.onload = (event) => {
+                const fileString = event.target.result as string;
+                setUploadedFileString(fileString);
+                props.setTextData(fileString);
+            };
+        } catch (error) {
+            console.error(
+                `Error parsing uploaded File as JSON: ${error.message}`
```

Comment:
I don't actually think any JSON parsing is happening in this function? It seems to me like the file contents are being read (assuming UTF-8 encoding) and then used to set state in the parent component. So at a minimum I think this error message needs to change.

Alternately, you could try connecting this to the parent component so that the "error `Alert`" appears with this `error.message`

Reply:
Changed failure banner's state to string to take in a custom error message as needed so it can be reused for different errors. Also changed the error message here